executor/executor.go

```diff
@@ -907,7 +907,8 @@ func (e *SelectLockExec) Next(ctx context.Context, req *chunk.Chunk) error {
 		return err
 	}
 	// If there's no handle or it's not a `SELECT FOR UPDATE` statement.
-	if len(e.tblID2Handle) == 0 || (e.Lock != ast.SelectLockForUpdate && e.Lock != ast.SelectLockForUpdateNoWait) {
+	if len(e.tblID2Handle) == 0 || (e.Lock.LockType != ast.SelectLockForUpdate &&
+		e.Lock.LockType != ast.SelectLockForUpdateNoWait && e.Lock.LockType != ast.SelectLockForUpdateWaitN) {
```

Comment:
extract a function then it can be used at https://github.com/pingcap/tidb/pull/19630/files#diff-f888b92d66ee2d478ea975b08cddaa6cR435